requests/models.py

```diff
@@ -16,6 +16,13 @@
 # such as in Embedded Python. See https://github.com/kennethreitz/requests/issues/3578.
 import encodings.idna
 
+try:
+    import idna as _idna
```

Comment:
So this is **absolutely** _unacceptable_.

While the problems in the `idna` module may not be resolved, the fact that we've recently switched away from vendoring *known* good versions means that users will still run into trouble with this if they have an old enough version installed with requests.

This import is conditionally inside the staticmethod because it is:

1. Not always needed
2. Potentially incredibly slow (on the order of seconds if I remember correctly) with older versions of idna